Extend next's documentation (#322)
crystal-lang/crystal-book: syntax_and_semantics/next.md

  end
  puts a
end

# The above prints the numbers 2, 4 and 5
```

`next` can also be used to exit from a block, for example:

```crystal
def block
  yield
end

block do

>   puts 1, 2

Reviewer: Please make this `puts` print only one value, it feels weird to complicate the example by making readers recall that puts can take multiple arguments and prints them on separate lines.

  end
  puts a
end

# The above prints the numbers 2, 4 and 5
```

> But `next` can also be used to exit from blocks, for example:

Reviewer 2: ```suggestion
`next` can also be used to exit from a block, for example:
```

I'm not sure if `exit` is a great term here, but I don't have a better idea.

```crystal
def block
  yield
end

block do
  puts 1, 2
  next
  puts 3
end

# The above prints the numbers 1 and 2
```

> Similar to [`break`](break.md), `next` can also take a parameter which will then be the value that gets returned:

Reviewer 2: ```suggestion
Similar to [`break`](break.md), `next` can also take a parameter which will then be returned by `yield`.
```

```crystal
def block
  yield
end

three = block do

>   puts 1, 2

Reviewer 2: This example cane be simplified. There is no need for a variable, just add `puts yield` directly in `block`. `puts 1, 2` is also unnecessary.

Author: The `puts 1, 2` is part of the example. When I would remove it then just printing `3` doesn't make much sense. I would probably make it `"hello"` then and then it would look pretty redundant and useless:
```crystal
def block
  puts yield
end

block do
  next "hello"
end
```
*why isn't this just `puts "hello"`?*

But when also having `puts 1, 2` and then printing the last number using `next`, this thought is resolved I think.

Reviewer 2: > why isn't this just puts "hello"?

Because the intention is to demonstrate the return value of `yield`. For this purpose you don't need to print `1, 2`. It's just unrelated noise.
Printing two numbers and then returning the next one using `next` doesn't have any useful meaning either.

Obviously, changing the value from `3` to a nice `"hello"` is better.
